src/main/java/com/salesforce/storm/spout/dynamic/DynamicSpout.java

```diff
@@ -162,16 +167,20 @@ public void open(Map topologyConfig, TopologyContext topologyContext, SpoutOutpu
         // Create MessageBus instance and store into SpoutMessageBus reference reducing accessible scope.
         this.messageBus = new MessageBus(messageBuffer);
 
-        // Create Spout SpoutCoordinator.
-        coordinator = new SpoutCoordinator(
-            // Our metrics recorder.
-            metricsRecorder,
-            // Pass message bus casted as a VirtualSpoutMessageBus.
-            (VirtualSpoutMessageBus) messageBus
+        // Define thread context, this allows us to use contextually relevant thread names.
+        final ThreadContext threadContext = new ThreadContext(
+            topologyContext.getThisComponentId(),
+            topologyContext.getThisTaskIndex()
         );
 
-        // Call open on coordinator, avoiding getter
-        coordinator.open(getSpoutConfig());
+        // Create Coordinator instance and call open.
+        spoutCoordinator = new SpoutCoordinator(
+            topologyConfig,
+            threadContext,
+            (VirtualSpoutMessageBus) messageBus,
```

Comment:
This smells... I can't put my finger on it, but I feel like having to do this here is a sign we could this better.

Reply:
I mean, its easily avoidable by using a local variable.